packages/lambdas/src/patient-discovery.ts

```diff
@@ -1,103 +1,42 @@
+import { processIncomingRequest } from "@metriport/core/external/carequality/pd/process-incoming-pd";
+import { MPIMetriportAPI } from "@metriport/core/mpi/patient-mpi-metriport-api";
+import { getEnvVarOrFail } from "@metriport/core/util/env-var";
+import { baseRequestSchema, PatientDiscoveryRequestIncoming } from "@metriport/ihe-gateway-sdk";
 import * as Sentry from "@sentry/serverless";
-import {
-  PatientDiscoveryRequestIncoming,
-  PatientDiscoveryResponseOutgoing,
-  baseRequestSchema,
-} from "@metriport/ihe-gateway-sdk";
-import { Patient } from "@medplum/fhirtypes";
+import { APIGatewayProxyEvent } from "aws-lambda";
 
-export const handler = Sentry.AWSLambda.wrapHandler(processRequest);
+// comment for diff...
+const apiUrl = getEnvVarOrFail("API_URL");
+const mpi = new MPIMetriportAPI(apiUrl);
 
-// Function to extract necessary fields and construct the responses
-async function processRequest(
-  payload: PatientDiscoveryRequestIncoming
-): Promise<PatientDiscoveryResponseOutgoing> {
-  // validate with zod schema
-  const baseRequest = baseRequestSchema.parse({
-    id: payload.id,
-    timestamp: payload.timestamp,
-    samlAttributes: payload.samlAttributes,
-    patientId: payload.patientId,
-  });
-
-  const fullRequest: PatientDiscoveryRequestIncoming = {
-    ...baseRequest,
-    patientResource: payload.patientResource,
-  };
-
-  if (Math.random() > 0.5) {
-    return constructErrorResponse(fullRequest);
+export const handler = Sentry.AWSLambda.wrapHandler(async (event: APIGatewayProxyEvent) => {
+  if (!event.body) {
+    return buildResponse(400, "Request body is missing");
```

Comment:
A string is not a valid JSON, so we prob want to wrap it around something like `{ message: "string" }`

Reply:
we call `JSON.stringify` in the buildResponse to make it a valid JSON

Reply:
I should have been clearer, sorry. First of all, a string is a valid JSON. What I meant was on the context of our API's response, which should be an object - [reference](https://www.notion.so/metriport/Development-Best-Practices-bb3574d3c83d46dc9914a9e5c5542ca7?pvs=4#f692f10d557d40f1b310b9896d41371f).

Reply:
Ok I see. I added the { message: "string" }